packages/nuxt/src/vite/sourceMaps.ts

```diff
@@ -87,25 +106,189 @@ export function getPluginOptions(
+        break;
+    }
+  } else {
+    if (nuxtSourcemap.client === false) {
+      warnExplicitlyDisabledSourcemap('sourcemap.client');
+      previousUserSourceMapSetting.client = 'disabled';
+    } else if (['hidden', true].includes(nuxtSourcemap.client)) {
+      logKeepSourcemapSetting(sentryModuleOptions, 'sourcemap.client', nuxtSourcemap.client.toString());
+      previousUserSourceMapSetting.client = 'enabled';
+    } else {
+      nuxt.options.sourcemap.client = 'hidden';
+      logSentryEnablesSourcemap('sourcemap.client', 'hidden');
+      previousUserSourceMapSetting.client = 'unset';
+    }
+
+    if (nuxtSourcemap.server === false) {
+      warnExplicitlyDisabledSourcemap('sourcemap.server');
+      previousUserSourceMapSetting.server = 'disabled';
+    } else if (['hidden', true].includes(nuxtSourcemap.server)) {
+      logKeepSourcemapSetting(sentryModuleOptions, 'sourcemap.server', nuxtSourcemap.server.toString());
+      previousUserSourceMapSetting.server = 'enabled';
+    } else {
+      nuxt.options.sourcemap.server = 'hidden';
+      logSentryEnablesSourcemap('sourcemap.server', 'hidden');
+      previousUserSourceMapSetting.server = 'unset';
+    }
+  }
+
+  return previousUserSourceMapSetting;
+}
+
+/** only exported for testing */
+export function changeViteSourcemapSettings(
+  viteConfig: { build?: { sourcemap?: boolean | 'inline' | 'hidden' } },
+  sentryModuleOptions: SentryNuxtModuleOptions,
+): UserSourcemapSetting {
+  viteConfig.build = viteConfig.build || {};
+  const viteSourcemap = viteConfig.build.sourcemap;
+
+  let previousUserSourceMapSetting: UserSourcemapSetting;
+
+  if (viteSourcemap === false) {
+    warnExplicitlyDisabledSourcemap('vite.build.sourcemap');
+    previousUserSourceMapSetting = 'disabled';
+  } else if (viteSourcemap && ['hidden', 'inline', true].includes(viteSourcemap)) {
+    logKeepSourcemapSetting(sentryModuleOptions, 'vite.build.sourcemap', viteSourcemap.toString());
+    previousUserSourceMapSetting = 'enabled';
+  } else {
+    viteConfig.build.sourcemap = 'hidden';
+    logSentryEnablesSourcemap('vite.build.sourcemap', 'hidden');
+    previousUserSourceMapSetting = 'unset';
+  }
+
+  return previousUserSourceMapSetting;
+}
+
+/** only exported for testing */
+export function changeRollupSourcemapSettings(
+  nitroConfig: {
+    rollupConfig?: {
+      output?: {
+        sourcemap?: OutputOptions['sourcemap'];
+        sourcemapExcludeSources?: OutputOptions['sourcemapExcludeSources'];
+      };
+    };
+  },
+  sentryModuleOptions: SentryNuxtModuleOptions,
+): UserSourcemapSetting {
+  nitroConfig.rollupConfig = nitroConfig.rollupConfig || {};
+  nitroConfig.rollupConfig.output = nitroConfig.rollupConfig.output || { sourcemap: undefined };
+
+  let previousUserSourceMapSetting: UserSourcemapSetting;
+
+  const nitroSourcemap = nitroConfig.rollupConfig.output.sourcemap;
+
+  if (nitroSourcemap === false) {
```

Comment:
l: the logic in here is basically the same for vite rollup and nuxt, maybe this could be moved into a function instead

Reply:
I thought about that as it would make the code eventually shorter. But I kept the three different functions (and so the duplication) because I think it adds more complexity if those functions are combined into one. The log outputs and the sourcmap option source are different for vite, rollup and nuxt and a function abstraction would make it very difficult to read. I think it's better to have some duplication here for the purpose of readability.